src/main/kotlin/org/rust/lang/core/psi/ext/RsPat.kt

```diff
@@ -0,0 +1,20 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.psi.ext
+
+import org.rust.lang.core.psi.*
+
+val RsPat.isRefutable: Boolean
+    get() = when (this) {
+        is RsPatStruct -> patFieldList.isNotEmpty() && patFieldList.any { it.pat?.isRefutable ?: false }
+        is RsPatTupleStruct -> patList.isNotEmpty() && patList.any { it.isRefutable }
+        is RsPatSlice -> patList.isNotEmpty() && patList.any { it.isRefutable }
+        is RsPatTup -> patList.isNotEmpty() && patList.any { it.isRefutable }
+        is RsPatRange, is RsPatConst -> true
+        is RsPatIdent -> parent is RsPat && parent !is RsPatTup
```

Comment:
Can you write a test for this case? I don't understand when a patident can be refutable.

Now the intention is available for this code:

```rust
match e {
    [x, y] => panic!(),
    _ => ()
};
```

But compiler yields this error:
```
  |
6 | if let [x, y] = e {
  |        ^^^^^^ irrefutable pattern
```

Reply:
My mistake. Fixed.